elastic/kibana
# [ESUI] Move XJson Mode to src/plugins/es_ui_shared

## Summary

`XJsonMode` provides Ace editors with all of the JSON capabilities in Console (including grammar checking).

It's great to see the adoption of XJSON in more places in Kibana now since it was extracted to a more shared location and made reusable. There were a few parts of the current code that needed cleaning up however.

Currently code is spread between the OSS es_ui_shared and the x-pack es_ui_shared. The OSS version also does not function correctly without the presence of the worker to interpret the grammar (the JSON worker does not understand XJSON)

This contribution moves all of the XJSON code to one place inside of es_ui_shared/public OSS (totally removes x-pack es_ui_shared).

Also, useXJson hook has also been moved to alongside the form hook lib and was expanded to accept objects and strings as input.

## How to review (@codewoners)

- Ensure that your syntax highlighting is still working as before
- Ensure the code changes are acceptable (in line with your patterns). Most changes are paths and code was mostly removed not changed

## Additional

Also converted files to TS and clean up use of @ts/ignore (focussed on ESUI code only).
